pkg/services/provisioning/datasources/config_reader_test.go

```diff
@@ -249,8 +250,10 @@ func TestDatasourceAsConfig(t *testing.T) {
 			}
 
 			require.Equal(t, 2, len(correlationsStore.created))
-			require.Equal(t, 0, len(correlationsStore.deletedBySourceUID))
```

Comment:
should we also test the new non-duplication logic?

Reply:
The provisioning test uses mocked data base so I'm not sure if there's a way to test it e2e. I've added a test at the service level to check if the method works as expected here https://github.com/grafana/grafana/pull/73737/files#diff-c64b433d85c39bd84c2b456f3da5dcb22c913e5ede528ab6b38c478b99621f2a.